### PR TITLE
Improvement: Avoid potential crashes and coding style

### PR DESCRIPTION
**Pull request by wutschel**
## Description

- Add boundary checks in EPG channel ID and (movie / episode / TV Show) cast processing.
- Use ternary operator in few places
- Improve naming of ivar (list of cast members)
- Remove `stringWithFormat` where not needed

## Summary for release notes

Improvement: Avoid potential crashes and coding style

### CODE REVIEW

File: XBMC Remote/ShowInfoViewController.m [hunk 1]
```diff
@@ -1535,16 +1535,19 @@ - (UITableViewCell*)tableView:(UITableView*)tableView cellForRowAtIndexPath:(NSI
...
+        NSDictionary *castMember = castList[indexPath.row];
+        NSString *serverURL = [Utilities getImageServerURL];
+        NSString *stringURL = [Utilities formatStringURL:castMember[@"thumbnail"] serverURL:serverURL];
+        [cell.actorThumbnail sd_setImageWithURL:[NSURL URLWithString:stringURL]
+                               placeholderImage:[UIImage imageNamed:@"person"]
+                                        options:SDWebImageScaleToNativeSize];
+        [Utilities applyRoundedEdgesView:cell.actorThumbnail drawBorder:YES];
+        cell.actorName.text = castMember[@"name"] ?: self.detailItem[@"label"];
+        if ([castMember[@"role"] length] != 0) {
+            cell.actorRole.text = [NSString stringWithFormat:@"%@", castMember[@"role"]];
```
Comment (kambala-decapitator) [inline XBMC Remote/ShowInfoViewController.m:1548]: can it be not a string?

**Review by kambala-decapitator — APPROVED** [review]

File: XBMC Remote/ShowInfoViewController.m (hunk 1 above)
Comment (wutschel) [reply XBMC Remote/ShowInfoViewController.m:1548]: Not really. Should I just remove `stringWithFormat`?

Comment (kambala-decapitator) [reply XBMC Remote/ShowInfoViewController.m:1548]: > Not really. Should I just remove `stringWithFormat`?
> 
> Even `cell.actorRole.text = castMember[@"role"]` w/o checking for `length` would be fine.

yes, you're right

Comment (wutschel) [reply XBMC Remote/ShowInfoViewController.m]: Done

**Pull request merged**